optimum/onnxruntime/trainer.py

```diff
@@ -260,13 +309,63 @@ def __init__(
             optimizers=optimizers,
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
         )
-
+        self._training_model = ModuleWithLoss(model, args)
```

Comment:
add a comment explaining how we leverage training_model and inference_model to do train and evaluate?

Reply:
Please also add comments explaining why we need to move from having just model to _training_model, _inferenceing_model and model... Does this mean we have multiple copies of the model now?